### PR TITLE
Feature/sttp stub test using

### PR DESCRIPTION
**Pull request by ghostbuster91**
This PR adds new API for testing endpoints. One can now call:

```scala
import sttp.tapir.server.stub._

val serverEndpoint = endpoint.serverLogic[F] { ... }
val response: Option[F[Either[Err, Result]]] = serverEndpoint.testUsing(someInput)
```
where `None` means that input matching has failed.

there is also a possibility to stub a whole set of endpoints:

```scala
import sttp.tapir.server.stub._

implicit val backend: SttpBackend[F, Nothing, NothingT]  = 
  List(endpoint1, endpoint2, ...).toBackendStub
```

### CODE REVIEW

**Comment by ghostbuster91** [comment]
Original discussion in https://github.com/softwaremill/tapir/pull/471

**Comment by kciesielski** [comment]
LGTM

File: server/sttp-stub-server/src/main/scala/sttp/tapir/server/stub/package.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import sttp.client.{NothingT, SttpBackend, _}
+import sttp.tapir.client.sttp._
+import sttp.tapir.server.internal.{DecodeInputs, DecodeInputsResult}
+
+package object stub extends SttpStubServer {
+
+  implicit class StubbedEndpoint[I, E, O, F[_]: MonadError](
+      serverEndpoint: ServerEndpoint[I, E, O, Nothing, F]
+  ) {
+    def testUsing(input: I): Option[F[Either[E, O]]] = {
```
Comment (adamw) [inline server/sttp-stub-server/src/main/scala/sttp/tapir/server/stub/package.scala]: weren't we suppose to remove this? I think we ended up deciding this doesn't test anything useful? :)

File: server/sttp-stub-server/src/main/scala/sttp/tapir/server/stub/SttpStubServer.scala [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+import sttp.tapir.server.internal.{DecodeInputs, DecodeInputsResult, InputValues}
+import sttp.tapir.server.{DecodeFailureContext, DecodeFailureHandling, ServerDefaults, ServerEndpoint}
+import sttp.tapir.{DecodeFailure, Endpoint, EndpointInput}
+
+trait SttpStubServer {
+
+  implicit class RichStubServerEndpoints[F[_]](
+      endpoints: Iterable[ServerEndpoint[_, _, _, _, F]]
+  ) {
+    def toBackendStub(implicit me: MonadError[F]): SttpBackend[F, Nothing, NothingT] = {
```
Comment (adamw) [inline server/sttp-stub-server/src/main/scala/sttp/tapir/server/stub/SttpStubServer.scala]: so there are two ways to use the stub now,yes? one using `.toBackendStub` on server endpoints, and the other using `when`/`then`?

File: server/sttp-stub-server/src/test/scala/sttp/tapir/server/stub/SttpStubServerTest.scala [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+      endpoint.serverLogic[Identity] { _ => Right(ResponseWrapper(0.3)) }
+
+    // when
+    val response = new StubbedEndpoint(endpointWithStubLogic).testUsing(-1)
+
+    // then
+    response shouldBe None
+  }
+
+  it should "stub an endpoint with passing validation" in {
```
Comment (adamw) [inline server/sttp-stub-server/src/test/scala/sttp/tapir/server/stub/SttpStubServerTest.scala]: there's still no test for endpoints with multiple inputs?

**Comment by adamw** [comment]
Docs are missing :)

File: server/sttp-stub-server/src/main/scala/sttp/tapir/server/stub/package.scala (hunk 1 above)
Comment (ghostbuster91) [reply server/sttp-stub-server/src/main/scala/sttp/tapir/server/stub/package.scala]: This PR is about `testUsing`.

**Comment by ghostbuster91** [comment]
Closed due to confusion spreading

**Pull request closed without merging**